### ISSUE TITLE
False positives in `no-empty-alternative` and `prefer-character-class` after updating eslint utils

### ISSUE THREAD

**Issue by kachkaev** [ota-meshi/eslint-plugin-regexp#828]
**Information:**
- ESLint version: 9.25.1
- `eslint-plugin-regexp` version: 2.7.0

**Description**

After updating transient dependencies, I noticed new violations of [`regexp/no-empty-alternative`](https://ota-meshi.github.io/eslint-plugin-regexp/rules/no-empty-alternative.html) and [`regexp/prefer-character-class`](https://ota-meshi.github.io/eslint-plugin-regexp/rules/prefer-character-class.html) rules.

This was related to updating [`@eslint-community/eslint-utils`](https://www.npmjs.com/package/@eslint-community/eslint-utils) from [4.6.1](https://github.com/eslint-community/eslint-utils/releases/tag/v4.6.1) to [4.7.0](https://github.com/eslint-community/eslint-utils/releases/tag/v4.7.0). Both versions are within the dependency range:

https://github.com/ota-meshi/eslint-plugin-regexp/blob/14ac3d8b0ca040c07c2c56966f77f38439267277/package.json#L122

This upstream PR looks relevant: https://github.com/eslint-community/eslint-utils/pull/255

Here is an MWE:

```js
const a = ["x", "y"] as const;
const b = ["x", "y", ""] as const;
const c = ["x", "y", "z"] as const;

const r1 = new RegExp(`^(${a.join("|")})$`);
const r2 = new RegExp(`^(${b.join("|")})$`);
const r3 = new RegExp(`^(${c.join("|")})$`);
```

Before `@eslint-community/eslint-utils` was updated, no errors were triggered for the above code. But now I see this:

```js
const r1 = new RegExp(`^(${a.join("|")})$`);
const r2 = new RegExp(`^(${b.join("|")})$`); // This empty alternative might be a mistake. If not, use a quantifier instead. eslint(regexp/no-empty-alternative)
const r3 = new RegExp(`^(${c.join("|")})$`); // Unexpected the disjunction of single element alternatives. Use character class '[...]' instead. eslint(regexp/prefer-character-class)
```

I believe that these are false positives. If a RegExp is built from the variables, it is impossible to meet the rules without introducing multiple sources of truth.

**Comment by ota-meshi** [comment]
I think the rule is working correctly, and because eslint-utils now understands `as const`, the rule work better.
If you have source code where the rule is giving a false positive, can you provide an example and a repository where it can be reproduced?
